components/layout/layout_task.rs

```diff
@@ -1460,6 +1583,105 @@ impl FragmentBorderBoxIterator for FragmentLocatingFragmentIterator {
     }
 }
 
+enum Side {
+    Left,
+    Right,
+    Bottom,
+    Top
+}
+
+enum MarginPadding {
+    Margin,
+    Padding
+}
+
+enum PositionProperty {
+    Left,
+    Right,
+    Top,
+    Bottom,
+    Width,
+    Height,
+}
+
+struct PositionRetrievingFragmentBorderBoxIterator {
```

Comment:
Since we're starting to get a bunch of these iterators, I wonder if we should move them to a different file—say, `queries.rs`.